Server/Image_module/contrast_stretching.py

```diff
@@ -0,0 +1,53 @@
+def contrast_stretching(id1,id2):
+    """
+        Process the input image with contrast_stretching and check the image size
+
+        :param id1: the input should be the name string of the image
+        :param id2: the input should be the name string of the processed image
+        :raises ImportError:  if input is not a string
+        :raises NameError:: if the id1 image file does not exist
+        :raises AssertionError: if the image array does not contain three layers
+
+        :returns: a python dictionary contains the base64 string of the processed image and the image size
+        :rtype: python dictionary
+        """
+    try:
+        import PIL
+        from PIL import Image
+        import numpy as np
+        from skimage import exposure
+        import os
+        from skimage import util
+        import logging
+        import uuid
+        from Image_module.encode_image import encode_image
+    except ImportError:
+        print("Necessary imports failed")
+    else:
+        logging.basicConfig(filename='histogram_equalization.log',
+                            level=logging.DEBUG, filemode='w')
+
+    try:
+        i = np.asarray(PIL.Image.open(id1))  # convert the image into the numpy array
+    except NameError:
+        logging.debug("The image file does not exist")
+        raise NameError
+
+    try:
+        assert len(i.shape) == 3            # Ensure the shape of the image array is right
+    except AssertionError:
+        logging.debug("The shape of image array is 3 layers")
```

Comment:
your assertion statement makes sure that `i.shape` is 3, so if there is an `AssertionError` that means that `i.shape` is **not** 3, so your logging statement should reflect that